src/main/kotlin/ecommerce/service/ProductService.kt

```diff
@@ -0,0 +1,53 @@
+package ecommerce.service
+
+import ecommerce.dto.ProductRequest
+import ecommerce.entity.OptionEntity
+import ecommerce.entity.ProductEntity
+import ecommerce.repository.ProductRepositoryJpa
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
+import org.springframework.stereotype.Service
+
+@Service
+class ProductService(
+    private val productRepositoryJpa: ProductRepositoryJpa,
+) {
+    fun createProduct(productRequest: ProductRequest): ProductEntity {
+        if (productRepositoryJpa.existsByName(productRequest.name)) {
+            throw IllegalArgumentException("Product with name '${productRequest.name}' already exists.")
+        }
```

Comment:
You did check for duplicate product names 👍

Reply:
But according to the requirement:

<img width="736" height="60" alt="image" src="https://github.com/user-attachments/assets/658ae669-7f0f-4f17-80fb-5a9764876283" />

<br/>
<br/>

It looks like you might need to check one more thing.